### ISSUE TITLE
why  Duration adverse loss don't  update

### ISSUE THREAD

**Issue by jiahong3837** [p0p4k/vits2_pytorch#3]
why  Duration adverse loss don't  update? the discrimatepredicton is compelete

**Comment by p0p4k** [comment]
I am just thinking if LSTM discriminator is right or some other transformer based discriminator is the way to go. It shouldnt really matter a lot, but just waiting on some feedback. Plus, if you think this LSTM discriminator is ok, please attach it to models.py and change lines in train.py and send a PR, it will be really helpful!

**Comment by jiahong3837** [comment]
你好，我是邱家洪，邮件已经收到，谢谢。

**Comment by p0p4k** [comment]
@elch10 I think the duration predictor is trained for the first 30k steps and then frozen. The overall waveform generation model is trained continuously for 800k steps.

**Comment by elch10** [comment]
I found such explanation in paper
`Our proposed duration predictor and training mechanism allow
for a learning duration in short steps, and the duration predictor
is separately trained as the last training step, which reduces the
overall computation time for training.`

**Comment by p0p4k** [comment]
I have added durations predictor to train all the way during the training. Will fix to follow the paper's style soon.

**Comment by JohnHerry** [comment]
> I have added durations predictor to train all the way during the training. Will fix to follow the paper's style soon.

The last 30K steps, will we just train the duration predictor with DD (duration discriminator)? or  train the whole model with DD?

**Comment by jiahong3837** [comment]
你好，我是邱家洪，邮件已经收到，谢谢。

**Comment by hammadmlk** [comment]
> I have added durations predictor to train all the way during the training. Will fix to follow the paper's style soon.

@p0p4k  I couldn’t find code for this in the training loop. 
is the expectation to manually change the config after the wave work generator model is trained?  
i.e load a checkpoint and use the new config for the last 30k steps?

**Comment by jiahong3837** [comment]
你好，邮件已经收到，谢谢。
